### ISSUE TITLE
Changing Realm data can only be done from inside a transaction

### ISSUE THREAD

**Issue by jahway603** [SchildiChat/SchildiChat-android#155]
**Describe the bug**
A friend signed up recently for an account on the Matrix.org server. I attempted to send them a direct message and received the following error message: "Changing Realm data can only be done from inside a transaction"

**To Reproduce**
1. Either search for user's name or click on it in a room.
2. Then attempt to Direct message them.
3. Then it fails and I get "Changing Realm data can only be done from inside a transaction"

**Expected behavior**
I should be able to successfully open a Direct message with them.

**Screenshots**
![schildichat-andr-error-huh](https://user-images.githubusercontent.com/64485701/195418070-adcf2c1b-9f3c-46f2-83b7-f4e2a01f91ce.png)

**Smartphone (please complete the following information):**
 - Device: [e.g. Samsung S6] Pixel 4a
 - OS: [e.g. Android 6.0] Android 13

**Additional context**
 - App version: 1.5.1.sc60
 - Store: F-Droid
 - Homeserver: matrix.org

**Upstream relevance**
 - Does this issue exist in Element Android? Not sure
 - If this issue exists in Element Android, please also link the corresponding bug report (create a new one if it doesn't exist yet)

**Comment by SpiritCroc** [comment]
Would be great if you can send a bug report with logs from the app after getting this error

**Comment by jahway603** [comment]
@SpiritCroc can definitely do that, but not sure what setting(s) need to change to do so. Do I need to enable "Verbose logs" or should I enable some option(s) under the "Schildi-debugging" section?

**Comment by SpiritCroc** [comment]
I don't think it should be necessary to enable any logging options. Verbose may help though, don't know right now.

**Comment by tna76874** [comment]
I confirm this bug on multiple devices. Deleting the cache is a temporary workaround - but after a while this error pops up again.

**Comment by sigma2017** [comment]
Hello, I am using a private server installed by me and I got the same error:
https://i.imgur.com/rSGGq5j.jpg
This is a Schildichat bug, on element android and element web this  error is not present.

**Comment by sigma2017** [comment]
As confirmed to Matrix room SchildiChat Android, the issue is resolved by disabling Settings / Labs / Enable deferred DMs.
Already tested, I was able to create a room with a new user, the error is not present anymore.
Thank you @SpiritCroc!

**Comment by SpiritCroc** [comment]
Would still be great if somebody experiencing this (on the latest beta release) could send in a bug report with logs :)

**Comment by mtris** [comment]
Hello: I just duplicated the error and sent a bug report by shaking my phone to submit. I hope that it sent what you requested. Thank you very much for your time.

**Comment by SpiritCroc** [comment]
> Hello: I just duplicated the error and sent a bug report by shaking my phone to submit. I hope that it sent what you requested. Thank you very much for your time.

Thank you! Unfortunately, there's not sufficient logging in v1.5.2, so I'm still waiting if this occurs on v1.5.8-based versions. There has been a new release yesterday (don't know when the official fdroid repo will have it), would be great if you can try again with that (or some recent beta).

**Comment by mtris** [comment]
Hello again: I updated to the v1.5.8, duplicated the error, and shook my phone to send a bug report. I hope this is useful. Thank you for your efforts and time. Sincerely,

**Comment by SpiritCroc** [comment]
> Hello again: I updated to the v1.5.8, duplicated the error, and shook my phone to send a bug report. I hope this is useful. Thank you for your efforts and time. Sincerely,

On first glance looks useful, thanks!
